### PR TITLE
feat(ci): Local Fastlane Setup for TestFlight

### LINKED ISSUE #82: feat(ci): Local Fastlane Setup for TestFlight
Closes #57

### PR DESCRIPTION
**Pull request by denrase**
# Overview

- App Updates
  - Added an app icon
  - Set app name to `Sentry Mobile`
  - Set bundle identifier to `io.sentry.mobile.app`

- Fastlane
  - Read build number from TestFlight and write incremented by one to `pubspec.yaml`
  - Get dependencies, clean and build for iOS release without code signing
  - Create IPA file
  - Upload IPA file

- Other
  - Updated `README.md` with instruction how to build and upload to TestFlight
  - Documented environmental variables needed for `fastlane`
  
Closes #82

### CODE REVIEW

**Comment by denrase** [comment]
Relates to #80

**Comment by denrase** [comment]
@bruno-garcia We have decided with @HazAT to move away form GitHub Actions and only provide the setup for automated release to TestFlight from a local machine.

**Comment by bruno-garcia** [comment]
Thanks @denrase makes sense to me

**Comment by bruno-garcia** [comment]
Sorry my ignorance but does this use our [Fastlane plugin](https://github.com/getsentry/sentry-fastlane-plugin) to get symbols uploaded to Sentry? I think it would be great to dogfood that with this app /cc @philipphofmann @HazAT

**Comment by denrase** [comment]
@bruno-garcia Not yet, i tied it out with another app and the sentry fastlane plugin, but couldn't download them immediatley after download, as the uplaoded app takes time for processing. But there are also some generated during the build, i will look into this, good call. 👍

File: README.md [hunk 1]
```diff
@@ -47,3 +47,22 @@ or run
 ```
 flutter run 
 ```
+
+6. Build iOS and upload to testflight
+
+Provide environment variables needed for fastlane. For example by updateing your `~/.bash_profile`
```
Comment (philipphofmann) [inline README.md:53]:
```suggestion
Provide environment variables needed for fastlane. For example by updating your `~/.bash_profile`
```

File: ios/.gitignore [hunk 2]
```diff
@@ -30,3 +30,6 @@ Runner/GeneratedPluginRegistrant.*
 !default.mode2v3
 !default.pbxuser
 !default.perspectivev3
+
+build/*
+fastlane/report.xml
```
Comment (philipphofmann) [inline ios/.gitignore:35]:
```suggestion
fastlane/report.xml

```

File: ios/fastlane/Fastfile [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+			xcargs: { # https://github.com/fastlane/fastlane/issues/12849#issuecomment-404978983
+				:BUNDLE_IDENTIFIER => app_identifier,
+				:PROVISIONING_PROFILE_SPECIFIER => "#{ENV['FASTLANE_PROVISIONING_PROFILE_NAME']}",
+				:DEVELOPMENT_TEAM => team_id,
+				:CODE_SIGN_IDENTITY => "iPhone Distribution"
+			},
+			export_options: {
+				method: "app-store",
+				provisioningProfiles: { 
+					app_identifier => "#{ENV['FASTLANE_PROVISIONING_PROFILE_NAME']}",
```
Comment (philipphofmann) [inline ios/fastlane/Fastfile:74]: I guess this should be app identifier?

**Comment by denrase** [comment]
@bruno-garcia I created a separate task for sentry fastlane step #98 🙇

**Pull request merged**